### PR TITLE
Selections controller

### PR DESCRIPTION
**Pull request by timparkyn**
@mmell Fixed selection controller and selection views

### CODE REVIEW

File: app/assets/stylesheets/app.scss
Comment (mmell) [inline app/assets/stylesheets/app.scss:100]: Usually you don't want to keep unused code around since it will pollute your code searches and add clutter.

File: app/controllers/users_controller.rb
Comment (mmell) [inline app/controllers/users_controller.rb:3]: This comment should read like this

    # FIXME: restrict for admin only

File: app/helpers/selections_helper.rb
Comment (mmell) [inline app/helpers/selections_helper.rb:11]: The instance variable should be created in the controller. This method should return the array. The controller should set `@forcasts` with this value.

File: app/helpers/selections_helper.rb
Comment (mmell) [inline app/helpers/selections_helper.rb:8]: The api key belongs in a config variable and accessed here with `Rails.configuration.wunderground_key`.

File: app/helpers/selections_helper.rb
Comment (mmell) [inline app/helpers/selections_helper.rb:6]: This `stations` variable is not used but the method call in the controller is doing the right thing by passing in `@selections`. Change this variable name to `selections` and use it on line 14.

File: app/helpers/selections_helper.rb
Comment (mmell) [inline app/helpers/selections_helper.rb:9]: This is not used either.

File: app/helpers/selections_helper.rb
Comment (mmell) [inline app/helpers/selections_helper.rb:13]: This comment seems out of date.

File: app/helpers/selections_helper.rb
Comment (mmell) [inline app/helpers/selections_helper.rb:14]: I think `each` could be `map` then you would not have to `@forecasts.push`. Just return the result of the `map`.

File: app/helpers/selections_helper.rb
Comment (mmell) [inline app/helpers/selections_helper.rb:17]: Always use symbols for hash keys unless you're forced to use strings.

Comment (mmell) [reply app/helpers/selections_helper.rb:17]: Why not
```
forecast = {
  selection_id: selection.id, 
  description: station.description
}
```

File: app/views/selections/_newStation.html.erb
Comment (mmell) [inline app/views/selections/_newStation.html.erb:1]: Remove extra blank lines. 
Indentation should always be exactly two spaces.
File names should be underscores (snake case?), not camel-case.

File: app/views/users/junk
Comment (mmell) [inline app/views/users/junk:1]: Remove `junk` file?

File: app/helpers/selections_helper.rb
Comment (mmell) [reply app/helpers/selections_helper.rb:14]: Change `current_user.selections` to `selections` (from the method call).

**Review by mmell — APPROVED** [review]
Approved, assuming you'll implement whatever suggestions make sense.

**Comment by mmell** [comment]
You're supposed to make the changes before merging! :)

**Comment by timparkyn** [comment]
Oops.  Was dealing with another issue (still am!).

I had merged another branch locally, while the pull request was outstanding.

So did ```git reset —hard <SHA>``` which seems to work.



So I was trying to simplify by merging the pull request, then I could merge my other branch, and then work on the issues you stated.

Can I undo the merge in Github?

**Comment by mmell** [comment]
No, don't undo it. It's not worth the headaches. I don't know how `git reset —hard <SHA>` would commit the PR though. But just carry on. Maybe address these comments in the next PR and mention this PR.

**Comment by timparkyn** [comment]
Did it an hour ago…. Refactoring based on your comments now

**Comment by timparkyn** [comment]
#mmell refactored all above except for api key... can't quite figure out how to do it.

**Comment by mmell** [comment]
Here's a good discussion of secrets handling in Rails: https://blog.arkency.com/2017/07/how-to-safely-store-api-keys-in-rails-apps/. For this round of changes the first option: "Save directly in codebase" is good enough. 

The second option "Save in ENV" is usually as far as we ever bother with so that's what I would consider the ideal. Be sure to look at the `dotenv` link. That's a common tool to make production code work the same in dev with minimal hassles.

**Comment by mmell** [comment]
Open a new PR for any more discussion.

**Pull request merged**
